rpcs3/Emu/Cell/Modules/cellRtc.cpp

```diff
@@ -3,298 +3,1672 @@
 
+	vm::var<sys_page_attr_t> page_attr;
+
+	error_code ret = sys_memory_get_page_attribute(pUtc.addr(), page_attr);
+	if (ret != CELL_OK)
+	{
+		return CELL_RTC_ERROR_INVALID_POINTER;
+	}
+
+	ret = sys_memory_get_page_attribute(pszDateTime.addr(), page_attr);
+	if (ret != CELL_OK)
+	{
+		return CELL_RTC_ERROR_INVALID_POINTER;
+	}
+
+	vm::var<CellRtcDateTime> date_time;
+
+	//	TODO fix unholyness
+
+	s64 lAdd;
+	u32 pcVar5;
+	u32 uVar7;
+	s32 iVar6;
+	s32 iVar8;
+
+	char cVar1 = pszDateTime[1];
+	char cVar2 = pszDateTime[2];
+	char cVar3 = pszDateTime[3];
+	if (((((((*pszDateTime - 0x30U) < 10) && ('/' < cVar1)) && (cVar1 < ':')) && (('/' < cVar2 && (cVar2 < ':')))) && ('/' < cVar3)) && (cVar3 < ':'))
+	{
+		date_time->year = (cVar2 << 1) + (cVar2 << 3) + *pszDateTime * 1000 + cVar1 * 100 + cVar3 + 0x2fb0;
+	}
+	else
+	{
+		date_time->year = 0xffff;
+	}
+
+	if (pszDateTime[4] != '-')
+	{
+		return CELL_RTC_ERROR_INVALID_YEAR;
+	}
+
+	cVar2 = pszDateTime[5];
+	cVar1 = pszDateTime[6];
+	if (((9 < (cVar2 - 0x30U)) || (cVar1 < '0')) || (date_time->month = ((cVar2 << 1) + (cVar2 << 3) + cVar1) - 0x210, '9' < cVar1))
+	{
+		date_time->month = 0xffff;
+	}
+
+	if (pszDateTime[7] != '-')
+	{
+		return CELL_RTC_ERROR_INVALID_MONTH;
+	}
+
+	cVar2 = pszDateTime[8];
+	cVar1 = pszDateTime[9];
+	if (((9 < (cVar2 - 0x30U)) || (cVar1 < '0')) || (date_time->day = ((cVar2 << 1) + (cVar2 << 3) + cVar1) - 0x210, '9' < cVar1))
+	{
+		date_time->day = 0xffff;
+	}
+
+	if (pszDateTime[10] != 'T' && pszDateTime[10] != 't')
+	{
+		return CELL_RTC_ERROR_INVALID_DAY;
+	}
+
+	cVar2 = pszDateTime[0xb];
+	cVar1 = pszDateTime[0xc];
+	if ((9 < (cVar2 - 0x30U)) || ((cVar1 < '0' || (date_time->hour = ((cVar2 << 1) + (cVar2 << 3) + cVar1) - 0x210, '9' < cVar1))))
+	{
+		date_time->hour = 0xffff;
+	}
+
+	if (pszDateTime[0xd] != ':')
+	{
+		return CELL_RTC_ERROR_INVALID_HOUR;
+	}
+
+	cVar2 = pszDateTime[0xe];
+	cVar1 = pszDateTime[0xf];
+	if (((9 < (cVar2 - 0x30U)) || (cVar1 < '0')) || (date_time->minute = ((cVar2 << 1) + (cVar2 << 3) + cVar1) - 0x210, '9' < cVar1))
+	{
+		date_time->minute = 0xffff;
+	}
+
+	if (pszDateTime[0x10] != ':')
+	{
+		return CELL_RTC_ERROR_INVALID_MINUTE;
+	}
+
+	cVar2 = pszDateTime[0x11];
+	cVar1 = pszDateTime[0x12];
+	if (((9 < (cVar2 - 0x30U)) || (cVar1 < '0')) || (date_time->second = ((cVar2 << 1) + (cVar2 << 3) + cVar1) - 0x210, '9' < cVar1))
+	{
+		date_time->second = 0xffff;
+	}
+
+	pcVar5 = 0x13;
+	if (pszDateTime[0x13] == '.')
+	{
+		date_time->microsecond = 0;
+		pcVar5                 = 0x14;
+		if ((pszDateTime[0x14] - 0x30U) < 10)
+		{
+			uVar7                  = 10000;
+			date_time->microsecond = (pszDateTime[pcVar5] + -0x30) * 100000;
+			while (true)
+			{
+				pcVar5 = pcVar5 + 1;
+				if (9 < (pszDateTime[pcVar5] - 0x30U))
+					break;
+				date_time->microsecond += uVar7 * (pszDateTime[pcVar5] + -0x30);
+				if (uVar7 == 100000)
+				{
+					uVar7 = 10000;
+				}
+				else
```

Comment:
this huge scoperino can be done with else ifs

Reply:
Honestly i don't even want to touch this block of code :|

Reply:
All you have to do is remove some scopes and use else if instead of if

Reply:
And to be honest.. all this code does is:
```
if (u_var_7 % 100 == 0)
{
    u_var_7 /= 10;
}
else
{
    u_var_7 = (u_var_7 ^ 10) - 1 >> 0x1f;
}
```